src/algebra/category/Mon/adjunctions.lean

```diff
@@ -23,6 +23,10 @@ universe u
 
 open category_theory
 
+-- typeclass inference cannot equate `with_one.monoid.to_mul_one_class` with
+-- `with_one.mul_one_class` without this.
+local attribute [semireducible] with_one with_zero
```

Comment:
This is the same problem as we faced with `tensor_algebra.ring`, where making the type irreducible makes unifying typeclasses about it fail.

Reply:
Okay. We can live with this.